File: epi_judge_python/test_framework/console_color.py

import sys
from enum import Enum, auto

from test_framework import platform


class ConsoleColor(Enum):
    FG_RED = auto()
    FG_GREEN = auto()
    FG_BLUE = auto()
    FG_YELLOW = auto()
    FG_DEFAULT = auto()


_color_codes_win = {
    ConsoleColor.FG_RED: 4 | 8, ConsoleColor.FG_GREEN: 2 | 8,
    ConsoleColor.FG_BLUE: 1 | 8, ConsoleColor.FG_YELLOW: 6 | 8,
    ConsoleColor.FG_DEFAULT: 7
}


def get_color_code_win(color):
    return _color_codes_win[color]


_color_codes_unix = {
    ConsoleColor.FG_RED: '\033[31m', ConsoleColor.FG_GREEN: '\033[32m',
    ConsoleColor.FG_BLUE: '\033[34m', ConsoleColor.FG_YELLOW: '\033[33m',
    ConsoleColor.FG_DEFAULT: '\033[39m'
}


def get_color_code_unix(color):
    return _color_codes_unix[color]


def print_std_out_colored(color, value):
    if not platform.use_color_output():
        print(value, end='')
        return

    if platform.running_on_win():
        old_text_attr = platform.get_console_screen_buffer_info().wAttributes

        sys.stdout.flush()
        platform.set_console_text_attribute(get_color_code_win(color))

        print(value, end='')

        sys.stdout.flush()
        platform.set_console_text_attribute(old_text_attr)
    else:
        print(
            '{}{}{}'.format(
                get_color_code_unix(color), value,
                get_color_code_unix(ConsoleColor.FG_DEFAULT)),
            end='')

File: epi_judge_java_solutions/epi/test_framework/ConsoleColor.java

package epi.test_framework;

public class ConsoleColor {
  public enum Color { FG_RED, FG_GREEN, FG_BLUE, FG_YELLOW, FG_DEFAULT }

  private static int getColorCodeWin(Color color) {
    switch (color) {
    case FG_RED:
      return 4 | 8;
    case FG_GREEN:
      return 2 | 8;
    case FG_BLUE:
      return 1 | 8;
    case FG_YELLOW:
      return 6 | 8;
    case FG_DEFAULT:
    default:
      return 7;
    }
  }

  private static String getColorCodeUnix(Color color) {
    switch (color) {
    case FG_RED:
      return "\033[31m";
    case FG_GREEN:
      return "\033[32m";
    case FG_BLUE:
      return "\033[34m";
    case FG_YELLOW:
      return "\033[33m";
    case FG_DEFAULT:
    default:
      return "\033[39m";
    }
  }

  public static void printStdOutColored(Color color, Object value) {
    if (!Platform.useColorOutput()) {
      System.out.print(value);
      return;
    }

    if (Platform.runningOnWin()) {
      System.out.flush();
      int oldTextAttr =
          Platform.winSetConsoleTextAttribute(getColorCodeWin(color));

      System.out.print(value);

      System.out.flush();
      Platform.winSetConsoleTextAttribute(oldTextAttr);
    } else {
      System.out.printf("%s%s%s", getColorCodeUnix(color),
                        String.valueOf(value),
                        getColorCodeUnix(Color.FG_DEFAULT));
    }
  }
}
